### ISSUE TITLE
Array(chn) drops third dimension for nchain==1

### ISSUE THREAD

**Issue by bgctw** [TuringLang/MCMCChains.jl#429]
This behavior of `Array(chn)` dropping the third dimension when only a single chain is present was unexpected to me. Is it meant to be this way?
It breaks code that uses `mapslices` on the Array over the third dimension, e.g. for computing an output for each sample, that does not check the special case of only a single chain before.

Minimal working example:
```
chn2 = Chains(rand(3, 2, 1), [:a, :b]); 
size(chn2)                  # 3-dimensional
size(Array(chn2))       # only 2-dimensional
```

**Comment by devmotion** [comment]
Yes, it is intended: https://github.com/TuringLang/MCMCChains.jl/blob/82def13b4ed4ae442d1fd655dfe3644b5113da74/src/constructors.jl#L54 By default, all chains are appended and the samples are returned as matrix of shape draws x params.

**Comment by devmotion** [comment]
If you specify `append_chains = false`, then instead a vector of matrices with the samples of each chain is returned.
